cmds/core/netstat/netstat_linux.go

```diff
@@ -0,0 +1,285 @@
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	flag "github.com/spf13/pflag"
+	"github.com/u-root/u-root/pkg/netstat"
+)
+
+var (
+	// Info source flags
+	routeFlag      = flag.BoolP("route", "r", false, "display routing table")
+	interfacesFlag = flag.BoolP("interfaces", "i", false, "display interface table")
+	ifFlag         = flag.StringP("interface", "I", "", "Display interface table for interface <if>")
+	groupsFlag     = flag.BoolP("groups", "g", false, "display multicast group memberships")
+	statsFlag      = flag.BoolP("statistics", "s", false, "display networking statistics (like SNMP)")
+
+	// Socket flags
+	tcpFlag  = flag.BoolP("tcp", "t", false, "TCP")
+	udpFlag  = flag.BoolP("udp", "u", false, "UDP")
+	udpLFlag = flag.BoolP("udplite", "U", false, "UDPlite")
+	rawFlag  = flag.BoolP("raw", "w", false, "RAW")
+	unixFlag = flag.BoolP("unix", "x", false, "UNIX")
+
+	// AF Flags
+	ipv4Flag = flag.BoolP("4", "4", false, "IPv4 flag. default: true")
+	ipv6Flag = flag.BoolP("6", "6", false, "IPv6 flag. default: false")
+
+	// Route type flag
+	routecacheFalg = flag.BoolP("cache", "C", false, "")
+
+	// Format flags
+	wideFlag      = flag.BoolP("wide", "W", false, "don't truncate IP addresses")
+	numericFlag   = flag.BoolP("numeric", "n", false, "don't resolve names")
+	numHostFlag   = flag.Bool("numeric-hosts", false, "don't resolve host names")
+	numPortsFlag  = flag.Bool("numeric-ports", false, "don't resolve port names")
+	numUsersFlag  = flag.Bool("numeric-users", false, "don't resolve user names")
+	symbolicFlag  = flag.BoolP("symbolic", "N", false, "resolve hardware names")
+	extendFlag    = flag.BoolP("extend", "e", false, "display other/more information")
+	programsFlag  = flag.BoolP("programs", "p", false, "display PID/Program name for sockets")
+	timersFlag    = flag.BoolP("timers", "o", false, "display timers")
+	continFlag    = flag.BoolP("continuous", "c", false, "continuous listing")
+	listeningFlag = flag.BoolP("listening", "l", false, "display listening server sockets")
+	allFlag       = flag.BoolP("all", "a", false, "display all sockets (default: connected)")
+)
+
+func evalFlags() error {
+	flag.Parse()
+
+	afs := make([]netstat.AddressFamily, 0)
+
+	// Validate info source flags
+	// none or one allowed allowed to be set
+	if !xorFlags(*routeFlag, *interfacesFlag, *groupsFlag, *statsFlag, *ifFlag != "") {
+		flag.Usage()
+		return nil
+	}
+
+	// Can't use default capability of pflags package, have to determine it like this
+	// to keep same usage as original netstat tool.
+	if !*ipv4Flag && !*ipv6Flag {
+		*ipv4Flag = true
+
+	}
+
+	// Why do I have to write it like that? It's ugly....MOM!
+	if (*ipv4Flag || *ipv6Flag || *statsFlag) &&
+		!(*tcpFlag || *udpFlag || *udpLFlag || *rawFlag || *unixFlag) {
+		*tcpFlag = true
+		*udpFlag = true
+		*udpLFlag = true
+		*rawFlag = true
+		*unixFlag = true
```

Comment:
why is there a unixFlag? 
Can't you just check runtime.GOOS? If you do, then when it is built, nice things happen. 
e.g. 
if runtime.GOOS == "linux" {
}
that code will not appear in the binary on non-Linux systems.

Reply:
unix flag triggers printing of unix sockets instead of IP sockets